---
title: "Introduction to R & RStudio"
subtitle: "Short-Lectures Zoom"
author: Lijiao NING
date: "2025-10-07"
format:
  revealjs: 
    history: false
    toc: true
    toc-title: "Overview"
    toc-depth: 1
    slide-number: true
    preview-links: auto
    logo: images/logoInforBio_fond_blanc_small.png
    css: custom.css
image: images/R_logo.png
embed-resources: true
execute: 
  echo: true
---

## üëã Aboute Me {.smaller}

Lijiao NING - Statistical Engineer (CNRS)

<div style="position:absolute; top:1em; right:3em;">
  ![](images/photo_ljn.png){width=150px style="border-radius:12px;"}
</div>

<br>

#### üéì Training & Work

- Master‚Äôs in biostatistics (2016 @University Lyon 1)
- European Genomic Institute for Diabetes (2016 - 2023 @Lille)
- Institute of Biology Paris-Seine (2023 - present)

#### üß¨ What I Do

- Analyze biological and multi-omics data (genotype, DNA methylation, RNA-seq, ChipSeq, *etc.*)
- Develop analysis pipelines in R
- Train biologists to use R for data analysis

:::{.notes}
Why I'm Here
- Help biologists turn data into insight
- Share practical R skills
- (Make R less scary and more fun!)
:::

# Why R for Biology? ![](images/R_logo.png){width=2.7cm}

## Welcome to R! {.smaller}

R is a programming language and software environment designed for statistical computing and data analysis.

::: columns
::: {.column}

Why R?

- Biologist friendly!
- Free and open-source
- Supported by a huge community

:::

::: {.column .fragment}

What you can do with R?

- Analyze your data
- Visualize your findings
- Automate your workflow
- Share reproducible research
:::
:::

The official website: <https://www.r-project.org>

## The R Console {.smaller}

::: columns
::: {.column}
![](images/r_console.png)
:::

::: {.column .fragment .incremental}
In the R console:

- **`>`** (The prompt): R is ready for your command.
Type your code and press **Enter** to run it.
- **`[1]`** (Output): This indicates the start of an output line.
- **`+`** (Continuation): R is waiting for you to finish your command.
You can either finish the command or press **Esc** to cancel it.
:::
:::

::: {.notes}
In the R Console, the > prompt lets you know that R is ready for your command. Just type your code and hit Enter.
When you see the output, it will start with [1], which indicates the position of the first element.
If you see a + at the beginning of a new line, it means R is waiting for you to finish your command,
this often happens if you forget a parenthesis or a comma.
You can either finish it by typing the rest of the code or press the Esc key to cancel the input completely and get a fresh > prompt.
:::


# Rstudio

## Rstudio {.smaller}

Integrated Development Environment (IDE)

::: columns
::: {.column width="55%"}
![](images/screenshot_rstudio.png)
:::

::: {.column width="45%" .fragment}
1. Console: enter and execute R commands interactively.
2. Environment/History: tracks your variables and past commands.
3. Files/Plots/Packages: managing your files, viewing graphs, or installing tools.

::: {.callout-note appearance="simple"}
Rstudio [cheat sheet](https://rstudio.github.io/cheatsheets/rstudio-ide.pdf){target="_blank"}
([and French version](https://rstudio.github.io/cheatsheets/translations/french/rstudio-ide_fr.pdf){target="_blank"})
:::
:::
:::

:::{.notes}
Everything you need in one organized window.
Save your work as a script, RStudio includes many useful features which let us code easily.
notebook, show results
:::


## Rstudio Login {.smaller}

If you use the RStudio Server, you need to connect to your account first.  

![](images/rstudio_server_login.png)

(Or if you are using IFB RStudio server, connect to your IFB account and create an RStudio session.)


## Rstudio Server {.smaller}

![](images/rstudio_server.png)

:::{.notes}
some buttons in addition with rstudio server...
:::

## Getting Started {.smaller}

::: columns

::: {.column width="43%"}

- Open your RStudio.
- Create an R script: *File* -> *New File* -> *R Script* or click ![](images/rstudio_newfile.png){width=1.5cm}. (Will appeared at 4)
- Write your codes in the script.
- Save script: *Cmd/Ctrl + S*, name your script.
- Execute code: Click ![](images/rstudio_runcode.png){width=1.3cm} or press *Cmd/Ctrl + Enter*

:::

::: {.column width="57%"}
![](images/rstudio_fullpanel.png)
:::
:::

## Getting Started {.smaller}

Copy paste following codes into your script and save the script as `test.R`.  

- Basic codes (commands):
``` {.r code-line-numbers="1|2-4|6-8|10-12|14-15|17-18|20-21"}
# Use `#` to write a comment (line) in your script
# Math with R:
2 + 2    # Yes, R can do math!
mean(c(1, 2, 3, 4, 5)) # Arithmetic mean.

# Create a variable:
x <- 5 # Now x is 5. You can reuse it!
x + 1

# Inspect your environment:
## Look at the "Environment" pane, do you see your x?
ls()

# Load a package:
library(stats)

# Get help of a function (documentation):
?mean

# Quit RStudio properly
q()
```

# Setting Up an R Project

## Create an R Project {.smaller}

An R project keeps everything for your project in **one working directory**
and helps R to know where to look for your files.

. . .

1. In RStudio, go to *File* -> *New Project*
2. Choose *New Directory* and click *New Project*
3. Set a folder name (e.g., `r_week1`) and select where to save it.
4. Click *Create Project*

. . .

You'll notice RStudio restarts, and now your project is set up!

![](images/in_rproj.png){width=50%}

::: {.notes}
An R Project is a way to keep everything for your work (scripts, data, outputs) in one place.
Think of it like a special folder where RStudio automatically knows where to look for your files.
It helps avoid problems with messy file paths.
:::


## Create an R Project {.smaller}

![](images/rstudio_workingdir.png)


## A Well Organized Project Folder {.smaller}

- Tree structure:

```sh
r_week1                 # Your working directory    
‚îú‚îÄ‚îÄ r_week1.Rproj       # The R project configuration file
‚îú‚îÄ‚îÄ README.md           # A short introduction of the project
‚îú‚îÄ‚îÄ data                # Raw data files
‚îÇ¬†¬† ‚îî‚îÄ‚îÄ penguins.csv
‚îú‚îÄ‚îÄ scripts             # Your R code files
‚îÇ¬†¬† ‚îî‚îÄ‚îÄ 01-analysis.R
‚îÇ¬†¬† ‚îî‚îÄ‚îÄ 02-figures.R
‚îî‚îÄ‚îÄ outputs             # Results, e.g., graphs, reports
```

. . .

- Managing Files (via the *File* pane):
  - To create a new folder: click ![](images/rstudio_create_folder_button.png){width=1cm}. The folder will be created in the current directory.
  - To move a file or folder: check the box next to its name, click ![](images/rstudio_gear.png){width=1cm} and then select *Move...* to choose a new destination.

. . .

::: {.callout-note appearance="simple"}
Check out the next short-lecture zoom: "Best practices in data science project management" (@Ivain Raslain)
:::


## How To Import Data?

::: columns
::: {.column style="font-size: 80%;"}

- Supported formats:  
  - text files (.txt, .csv, *etc.*)
  - Excel files
  - data from other softwares (*e.g.*: SAS)

- Import by clicking: 
  - Menu bar *File* -> *Import Dataset* -> Select the format -> Choose your file
:::

::: {.column .fragment}
![](images/import_data_base.png)
:::
:::


## Understanding File Paths {.smaller}

How to tell your computer where to find the `penguins.csv`?

![](images/finder_abs_path.png){.nostretch}

. . .

- **Absolute path**, starts from your computer's root (top-level) folder.
*E.g.*: 
  - Linux/Mac: `/Users/lijiao/Documents/r_week1/data/penguins.csv`
  - Windows: `C:\Users\lijiao\Documents\r_week1\data\penguins.csv`

. . .

```r
read.csv("/Users/lijiao/Documents/r_week1/data/penguins.csv")
```

## Understanding File Paths {.smaller}

A **path** tells your computer where to find files/folders.

- **Relative path**, starts from your working directory (folder). *E.g.*:
  - Linux/Mac: `data/penguins.csv` 
  - Windows: `data\penguins.csv`

![](images/finder_rel_path.png){width=80% .nostretch}

. . .

```r
read.csv("data/penguins.csv")
```

# A Simple Workflow

## Import Data {.smaller}

We'll use the [palmerpenguins](https://allisonhorst.github.io/palmerpenguins/articles/intro.html) dataset,
which contains different measurements of three species of penguins from three islands.
(You can get the raw data [here](https://raw.githubusercontent.com/allisonhorst/palmerpenguins/refs/heads/main/inst/extdata/penguins.csv){target="_blank"}.)

```{r}
# import data using url
penguins <- read.csv("https://raw.githubusercontent.com/allisonhorst/palmerpenguins/refs/heads/main/inst/extdata/penguins.csv")

# preview the first lines
head(penguins)
```

## Summary Statistics {.smaller}

```{r}
# data dimension
dim(penguins)

# summary statistics
summary(penguins)
```

## Visualisation {.smaller}

::: {.columns}
::: {.column}
```{r}
#| eval: false
#| warning: false

library(ggplot2)

ggplot(
  data = penguins,
  aes(
    x = bill_length_mm,
    y = flipper_length_mm,
    color = species
  )
) +
  geom_point()
```
:::

::: {.column}
```{r}
#| echo: false
#| fig-height: 10

library(ggplot2)

ggplot(data = penguins, aes(x = bill_length_mm, y = flipper_length_mm, color = species)) +
  geom_point() +
  theme(
    axis.title = element_text(size = rel(2)),
    axis.text = element_text(size = rel(2)),
    strip.text = element_text(size = rel(2)),
    plot.title = element_text(size = rel(2.5)),
    legend.text = element_text(size = rel(2)),
    legend.title = element_text(size = rel(2))
  )
```
:::
:::


## Generate a Report {.smaller}

Let's generate a quick report using Quarto ;)

![](images/demo_quarto.png)


# Interactive Online Companionship (IOC)

## IOC R and scRNAseq

- Technical trainings in small groups (in English and/or in French)
  - **R for data science for the beginners** (November to January)
  - **Single cell RNAseq** (March 2026)
- 10 sessions of 3 hours with 3 in-person
- Personalized follow-up

<div style="position:absolute; bottom:1em; right:1em;">
  ![More About IOC ‚òùÔ∏è](images/qrcode_info_ioc.png){width=250px style="border-radius:12px;"}
</div>


## Take Home Messages

- R + RStudio = free, powerful, reproducible
- Upcoming IOC **R for data science** (November to January) and **single cell RNAseq** (March to June)!

<div style="position:absolute; bottom:2em; right:3em;">
  ![IOC Registration ‚òùÔ∏è](images/qrcode_ioc.png){width=250px style="border-radius:12px;"}
</div>